enhancements/baremetal-IPI/baremetal.md

```diff
@@ -0,0 +1,171 @@
+  - "@sadasu"
+reviewers:
+  - "@smarterclayton"
+  - "@abhinavdahiya"
+  - "@enxebre"
+  - "@deads2k"
+approvers:
+  - "@abhinavdahiya"
+  - "@smarterclayton"
+  - "@enxebre"
+  - "@deads2k"
+creation-date: 2019-11-06
+last-updated: 2019-11-06
+status: implemented
+---
+
+# Adding Baremetal IPI capabilities to OpenShift
+
+This enhancement serves to provide context for a whole slew of features
+and enhancements that will follow to make Baremetal IPI deployments via
+OpenShift a reality.
+
+At the time of this writing, code for some of these enhancements have already
+merged, some are in progress and others are yet to me implemented. References
+to all these features in different stages of development will be provided
+below.
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+Baremetal IPI deployments enable OpenShift to enroll baremetal servers to become
+Nodes that can run K8s workloads.
+The Baremetal Operator [1] along with other provisioning services (Ironic and
+dependencies) run in their own pod called "metal3". This pod is deployed by the
+Machine API Operator when the Platform type is `BareMetal`. The OpenShift
+Installer is responsble for providing all the necessary configs required for
+a successful deployment.
+
+## Motivation
+
+The motivation for this enhancement request is to provide a background for all the
+the subsequent enhancement requests for Baremetal IPI deployments.
+
+### Goals
+
+The goal of this enhancement request is to provide context for all the changes
+that have already been merged towards making Baremetal IPI deployments a reality.
+All future Baremetal enhancement requests will refer back to this one to provide
+context.
+
+### Non-Goals
+
+Raising development PRs as a result of this enhancement request.
+
+## Proposal
+
+Every OpenShift based Baremetal IPI deployment will run a "metal3" pod on
+one Master Node. A "metal3" pod includes a container running BareMetal
+Operator(BMO) and several other supporting containers that work together.
+
+The BMO and other supporting containers together are able to discover a
+baremetal server in a pre-determined provisioning network, learn the
+HW attributes of the server and eventually boot it to make it available
+as a Machine within a MachineSet.
+
+The Machine API Operator (MAO) currently deploys the "metal3" pod only
+when the Platform type is `BareMetal` but installs the BaremetalHost CRD
+for all platform types starting from release 4.2.
+
+The installer is responsible for kicking off a Baremetal IPI deployment
+with the right configuration.
+
+A set of follow-up enhancement requests will try to bring in the following
+enhancements:
+1. Providing the BMO and it supporting containers with configuration that
```

Comment:
typo it/its

Reply:
I'd probably just keep this to represent current state of the world so there's no confusion this enhancements are discussed separately

Reply:
Agreed.